### PR TITLE
chore: use github action

### PR DESCRIPTION
**Pull request by pei0804**
**Describe the PR**
Benefits of using GitHubAction
- Jobs running in 20 of parallel.
- GitHubAPI will be easier to use.
- Matrix tests are easy to write.
- The job ends earlier than Travis.

### CODE REVIEW

**Comment by pei0804** [comment]
I don't have the right of codecov, so I would like you to set it up in another PR.
If you set up token, I can do it.

```
        - name: Upload Coverage report to CodeCov
          uses: codecov/codecov-action@v1.0.0
          with:
            token: ${{secrets.CODECOV_TOKEN}}
            file: ./profile.out
```
https://github.com/marketplace/actions/codecov

**Comment by pei0804** [comment]
@easonlin404 
review please.

File: Makefile [hunk 1]
```diff
@@ -59,7 +59,7 @@ deps:
 
 .PHONY: devel-deps
 devel-deps:
-	GO111MODULE=off $(GOGET) -v ${u} \
+	GO111MODULE=off $(GOGET) -v -u \
```
Comment (pei0804) [inline Makefile]: dev tool is want to use latest.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+on: push
+jobs:
+  test:
+    strategy:
+      matrix:
+        go: [ '1.10.x', '1.11.x', '1.12.x' ]
+        platform: [ubuntu-latest, macos-latest]
```
Comment (pei0804) [inline .github/workflows/ci.yaml]: You can increase of Windows tests, but the tests will fail.

Comment (easonlin404) [reply .github/workflows/ci.yaml]: What’re causes of failure?

Comment (pei0804) [reply .github/workflows/ci.yaml]: This is a failed reason.
https://github.com/swaggo/swag/pull/528/checks?check_run_id=232011601

**Comment by pei0804** [comment]
@easonlin404 
commented

**Comment by appleboy** [comment]
@pei0804 Please rename the `ci.yaml` to `ci.yml`. GitHub can't load the extension with `.yaml`

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+on: push
+jobs:
+  test:
+    strategy:
+      matrix:
+        go: [ '1.10.x', '1.11.x', '1.12.x' ]
```
Comment (appleboy) [inline .github/workflows/ci.yaml]: also add `1.13.x`

Comment (pei0804) [reply .github/workflows/ci.yaml]: We can't support for 1.13.x and I don't understand fix.
https://github.com/swaggo/swag/pull/507

**Comment by pei0804** [comment]
> @pei0804 Please rename the ci.yaml to ci.yml. GitHub can't load the extension with .yaml
OK!

**Comment by easonlin404** [comment]
> I don't have the right of codecov, so I would like you to set it up in another PR.
> If you set up token, I can do it.
> 
> ```yaml
>         - name: Upload Coverage report to CodeCov
>           uses: codecov/codecov-action@v1.0.0
>           with:
>             token: ${{secrets.CODECOV_TOKEN}}
>             file: ./profile.out
> ```
> 
> https://github.com/marketplace/actions/codecov

Seems like don't need to setup if it's public repo, right?

**Comment by pei0804** [comment]
@easonlin404 

https://github.com/swaggo/swag/pull/528#issuecomment-534606285
`token: ${{secrets.CODECOV_TOKEN}}`

I want to give me token.
https://github.com/marketplace/actions/codecov#usage

**Pull request merged**